rotkehlchen/chain/ethereum/uniswap/uniswap.py

```diff
@@ -106,6 +119,90 @@ def __init__(
+        pool_balance: Dict[ChecksumEthAddress, LiquidityPool] = (
+            {pool.address: pool for pool in balances}
+        )
+        # quick lookup, `agg` from aggregated
+        pool_events_agg_balance: Dict[ChecksumEthAddress, Dict[str, Any]] = {}
+        # Populate `pool_events_agg_balance` dict, being the keys the pools'
+        # addresses and the values their aggregated balances from their events
+        for event in events:
+            pool = event.pool_address
+
+            if pool not in pool_events_agg_balance:
+                # Default dictionary for amounts aggregation
+                pool_events_agg_balance[pool] = {
+                    'events': [],
+                    'profit_loss0': ZERO,
+                    'profit_loss1': ZERO,
+                    'usd_profit_loss': ZERO,
+                    'lp_profit_loss': ZERO,
+                }
+
+            pool_events_agg_balance[pool]['events'].append(event)
+
+            if event.event_type == EventType.MINT:
+                pool_events_agg_balance[pool]['profit_loss0'] += FVal(event.amount0)
+                pool_events_agg_balance[pool]['profit_loss1'] += FVal(event.amount1)
+                pool_events_agg_balance[pool]['usd_profit_loss'] += FVal(event.usd_price)
+                pool_events_agg_balance[pool]['lp_profit_loss'] += FVal(event.lp_amount)
+            else:  # event_type == EventType.BURN
+                pool_events_agg_balance[pool]['profit_loss0'] -= FVal(event.amount0)
+                pool_events_agg_balance[pool]['profit_loss1'] -= FVal(event.amount1)
+                pool_events_agg_balance[pool]['usd_profit_loss'] -= FVal(event.usd_price)
+                pool_events_agg_balance[pool]['lp_profit_loss'] -= FVal(event.lp_amount)
+
+        # Instantiate `LiquidityPoolEventsBalance` per pool using
+        # `pool_events_agg_balance`. If `pool_balance` exist (all events case),
+        # factorise in the current pool balances in the totals.
+        for pool, events_agg_balance in pool_events_agg_balance.items():
+            profit_loss0 = events_agg_balance['profit_loss0']
+            profit_loss1 = events_agg_balance['profit_loss1']
+            usd_profit_loss = events_agg_balance['usd_profit_loss']
+            lp_profit_loss = events_agg_balance['lp_profit_loss']
+
+            # Aggregate current pool balances looking up the pool
+            if pool in pool_balance:
+                token0 = pool_balance[pool].assets[0].asset
+                token1 = pool_balance[pool].assets[1].asset
+                profit_loss0 -= FVal(pool_balance[pool].assets[0].user_balance.amount)
+                profit_loss1 -= FVal(pool_balance[pool].assets[1].user_balance.amount)
+                usd_profit_loss -= FVal(pool_balance[pool].user_balance.usd_value)
+                lp_profit_loss -= FVal(pool_balance[pool].user_balance.amount)
+            else:
+                # NB: get `token0` and `token1` from any pool event
```

Comment:
what does NB mean?

Reply:
Sorry... XD. NB stands for "Nota Bene"; an important comment, like a "be aware".

https://en.wiktionary.org/wiki/nota_bene#:~:text=nota%20bene%20(plural%20notate%20bene,or%20warning%20to%20a%20text.
https://en.wikipedia.org/wiki/Nota_bene

![image](https://user-images.githubusercontent.com/14769962/99525169-5e0f7480-2991-11eb-91eb-0837637f018a.png)

Reply:
Hoooo ... thanks. I did not know of this abbreviation :smile: